### PR TITLE
Fixing GCC+Clang compile warnings

### PR DESCRIPTION
**Pull request by zpalmtree**
Code now compiles with zero warnings on clang 8 + gcc 9, on Linux.

### CODE REVIEW

**Comment by brandonlehmann** [comment]
CI test failed so I’m calling your bluff.

File: src/cryptonotecore/Currency.cpp [hunk 1]
```diff
@@ -289,8 +288,8 @@ namespace CryptoNote
         uint64_t summaryAmounts = 0;
         for (size_t no = 0; no < outAmounts.size(); no++)
         {
-            Crypto::KeyDerivation derivation = boost::value_initialized<Crypto::KeyDerivation>();
```
Comment (brandonlehmann) [inline src/cryptonotecore/Currency.cpp]: There's one more of these in there that I think can come easily out, mind grabbing it? (L#57)

While you're in there... the trim on L#475, if that can be removed I believe will remove the boost requirement from this file.

Comment (zpalmtree) [reply src/cryptonotecore/Currency.cpp]: Done

Comment (brandonlehmann) [reply src/cryptonotecore/Currency.cpp]: Thanks sweaty.

**Pull request merged**